src/ArborX_BruteForce.hpp

```diff
@@ -0,0 +1,124 @@
+/****************************************************************************
+ * Copyright (c) 2012-2019 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_BRUTE_FORCE_HPP
+#define ARBORX_BRUTE_FORCE_HPP
+
+#include <ArborX_AccessTraits.hpp>
+#include <ArborX_Box.hpp>
+#include <ArborX_CrsGraphWrapper.hpp>
+#include <ArborX_DetailsBruteForceImpl.hpp>
+#include <ArborX_DetailsKokkosExtAccessibilityTraits.hpp>
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX
+{
+
+template <typename MemorySpace>
+class BruteForce
+{
+public:
+  using memory_space = MemorySpace;
+  static_assert(Kokkos::is_memory_space<MemorySpace>::value, "");
+  using size_type = typename MemorySpace::size_type;
+  using bounding_volume_type = Box;
+
+  BruteForce() = default;
+
+  template <typename ExecutionSpace, typename Primitives>
+  BruteForce(ExecutionSpace const &space, Primitives const &primitives);
+
+  KOKKOS_FUNCTION
+  size_type size() const noexcept { return _size; }
+
+  KOKKOS_FUNCTION
+  bool empty() const noexcept { return size() == 0; }
+
+  KOKKOS_FUNCTION
+  bounding_volume_type bounds() const noexcept { return _bounds; }
+
+  template <typename ExecutionSpace, typename Predicates, typename Callback,
+            typename Ignore = int>
```

Comment:
What is the point of the `Ignore` template argument? Is it for SFINAE with other `query` function? If yes, why not SFINAE on `!Kokkos::is_view`?